pkgs/io_file/test/copy_file_test.dart

```diff
@@ -20,322 +20,324 @@ import 'package:win32/win32.dart' as win32;
 
 import 'errors.dart' as errors;
 import 'fifo.dart';
+import 'file_system_file_utils.dart' hide fileUtils;
 import 'test_utils.dart';
 
-void main() {
-  group('copyFile', () {
-    late String tmp;
-    late String cwd;
+void tests(FileUtils utils, FileSystem fs) {
+  late String tmp;
+  late String cwd;
 
-    setUp(() {
-      tmp = createTemp('copyFile');
-      cwd = fileSystem.currentDirectory;
-      fileSystem.currentDirectory = tmp;
-    });
+  setUp(() {
+    tmp = utils.createTestDirectory('createDirectory');
+    cwd = fs.currentDirectory;
+    fs.currentDirectory = tmp;
+  });
+
+  tearDown(() {
+    fs.currentDirectory = cwd;
+    utils.deleteDirectoryTree(tmp);
+  });
 
-    tearDown(() {
-      fileSystem.currentDirectory = cwd;
-      deleteTemp(tmp);
+  test('copy does not preserve source permissions', () {
+    final data = randomUint8List(1024);
+    final oldPath = '$tmp/file1';
+    final newPath = '$tmp/file2';
+    utils.createBinaryFile(oldPath, data);
+
+    // Give "other users" write permissions.
+    using((arena) {
+      // rw-rw-rw-
+      if (libc.chmod(oldPath.toNativeUtf8(allocator: arena).cast(), 438) ==
```

Comment:
Nit: 0666. File permissions are the one place octel literals are nicer.

Reply:
Sadly, dart does not support that ;-)